### PR TITLE
cemu: 2.0-65 -> 2.0-66, pin imgui 1.90.1

### LINKED ISSUE #289416: Build failure: cemu
### Steps To Reproduce

1. build *cemu*

### Build log

```
error: builder for '/nix/store/mxmjp17sfaa99r10vfvda34yp25vn5xd-cemu-2.0-65.drv' failed with exit code 2;
       last 10 log lines:
       > [ 18%] Building CXX object src/imgui/CMakeFiles/imguiImpl.dir/imgui_impl_opengl3.cpp.o
       > [ 18%] Building CXX object src/imgui/CMakeFiles/imguiImpl.dir/__/__/dependencies/imgui/imgui_widgets.cpp.o
       > /build/source/src/imgui/imgui_extension.cpp: In function 'void ImGui_UpdateWindowInformation(bool)':
       > /build/source/src/imgui/imgui_extension.cpp:133: error: 'struct ImGuiIO' has no member named 'ImeWindowHandle'
       >   133 |         io.ImeWindowHandle = nullptr;
       >       |
       > make[2]: *** [src/imgui/CMakeFiles/imguiImpl.dir/build.make:125: src/imgui/CMakeFiles/imguiImpl.dir/imgui_extension.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:703: src/imgui/CMakeFiles/imguiImpl.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/mxmjp17sfaa99r10vfvda34yp25vn5xd-cemu-2.0-65.drv'.
```

### Additional context

In ImGui changelog for 1.90.2:
`Commented out ImGuiIO::ImeWindowHandle obsoleted in 1.87 in favor of writing to void* ImGuiViewport::PlatformHandleRaw`

### Notify maintainers

cc @zhaofengli @baduhai

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240215.a4d4fe8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by qubitnano**
## Description of changes

Changelog: https://github.com/cemu-project/Cemu/compare/v2.0-65...v2.0-66

Fixes https://github.com/NixOS/nixpkgs/issues/289416

cemu doesn't build with imgui 1.90.2 or newer, so pin to 1.90.1 for now.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rowanG077** [comment]
Can confirm it now builds successfully. @SuperSandro2000

File: pkgs/applications/emulators/cemu/default.nix [hunk 1]
```diff
@@ -32,7 +32,19 @@
 , nix-update-script
 }:
 
-stdenv.mkDerivation rec {
+let
+  #cemu 2.0-65 doesn't build with imgui 1.90.2 or newer
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/cemu/default.nix:36]:
```suggestion
  # cemu 2.0-65 doesn't build with imgui 1.90.2 or newer:
  # error: 'struct ImGuiIO' has no member named 'ImeWindowHandle'
```

File: pkgs/applications/emulators/cemu/default.nix [hunk 2]
```diff
@@ -32,7 +32,19 @@
...
+  imgui' = imgui.overrideAttrs {
+    version = "1.90.1";
+    src = fetchFromGitHub {
+      owner = "ocornut";
+      repo = "imgui";
+      rev = "v1.90.1";
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/cemu/default.nix:42]:
```suggestion
  imgui' = imgui.overrideAttrs rec {
    version = "1.90.1";
    src = fetchFromGitHub {
      owner = "ocornut";
      repo = "imgui";
      rev = "v${version}";
```

**Comment by romatthe** [comment]
Still builds fine on 2.0-66 

1 package built:
cemu

Result of `nixpkgs-review pr 289867` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cemu</li>
  </ul>
</details>

**Comment by rowanG077** [comment]
If we are changing anything here would it be possible to add `aarch64-linux` to supported platforms? I tested it and it compiles and runs.

**Comment by baduhai** [comment]
Cemu might compile and even run on aarch64, but as far as I know, games don't run on aarch64 due to cemu's powerpc to x86 translator. See more here: https://wiki.cemu.info/wiki/Roadmap#LLVM_as_a_CPU_JIT_backend

**Review by baduhai — APPROVED** [review]
LGTM

**Review by marsam — APPROVED** [review]
Thank you!

**Pull request merged**